tendermint/src/rpc/id.rs

```diff
@@ -18,12 +26,16 @@ impl Id {
             .set_version(uuid::Version::Random)
             .build();
 
-        Id(uuid.to_string())
+        Id::Str(uuid.to_string())
     }
 }
 
 impl AsRef<str> for Id {
     fn as_ref(&self) -> &str {
-        self.0.as_ref()
+        match self {
+            Id::Num(_) => "",
```

Comment:
So if this is a number `as_ref` is supposed to return an empty string? Is this used somewhere? Maybe add a TODO here (or capture this in an issue, e.g #88 or a dedicated one).

Reply:
I think, this AsRef doesn't make much sense with the `enum`. AsRef trait has some interesting requirements: [doc](https://doc.rust-lang.org/std/convert/trait.AsRef.html)
- This trait must not fail. So, I need to return something.
- Used to do a cheap reference-to-reference conversion. So, I shouldn't use new memory.

This means, I can't do much with a number. The reutrn type is `str`, which means it would make sense to convert the number into its string representation but then it shouldn't be done in AsRef because it requires new memory. The documentation suggests: If you need to do a costly conversion it is better to implement From with type &T or write a custom function.

So, we should just get rid of it.

Reply:
(actually, maybe using a generic helps...)

Reply:
Got rid of it.